activerecord/lib/active_record/associations/association_scope.rb

```diff
@@ -135,13 +135,15 @@ def add_constraints(scope, owner, chain)
 
               if scope_chain_item == chain_head.scope
                 scope.merge! item.except(:where, :includes, :unscope, :order)
-              elsif !item.references_values.empty?
```

Comment:
`references_values` is populated, for example, when [passing a hash to where clauses](https://github.com/rails/rails/blame/4afb7249c4fa7f08ce81bb143c9882996e173ab9/activerecord/lib/active_record/relation/query_methods.rb#L1637) but not when passing a string. This should not determine if an explicit join is added to the relation.